packages/plugin-session-replay-browser/src/session-replay.ts

```diff
@@ -61,48 +74,65 @@ class SessionReplay implements SessionReplayEnrichmentPlugin {
   }
 
-  async emptyStoreAndReset() {
+  async initialize(shouldSendStoredEvents = false) {
+    this.timeAtLastSend = Date.now(); // Initialize this so we have a point of comparison when events are recorded
+    if (!this.config.sessionId) {
+      return;
+    }
     const storedReplaySessions = await this.getAllSessionEventsFromStore();
-    if (storedReplaySessions) {
-      for (const sessionId in storedReplaySessions) {
-        const storedReplayEvents = storedReplaySessions[sessionId];
-        if (storedReplayEvents.events.length) {
+    const storedSequencesForSession = storedReplaySessions && storedReplaySessions[this.config.sessionId];
+    if (storedReplaySessions && storedSequencesForSession && storedSequencesForSession.sessionSequences) {
+      const storedSeqId = storedSequencesForSession.currentSequenceId;
+      const lastSequence = storedSequencesForSession.sessionSequences[storedSeqId];
+      if (lastSequence.status !== RecordingStatus.RECORDING) {
+        this.currentSequenceId = storedSeqId + 1;
+        this.events = [];
+      } else {
+        // Pick up recording where it was left off in another tab or window
+        this.currentSequenceId = storedSeqId;
+        this.events = lastSequence.events;
+      }
+    }
+    if (shouldSendStoredEvents && storedReplaySessions) {
+      this.sendStoredEvents(storedReplaySessions);
+    }
+    if (!this.stopRecordingEvents) {
+      this.recordEvents();
+    }
+  }
+
+  sendStoredEvents(storedReplaySessions: IDBStore) {
+    for (const sessionId in storedReplaySessions) {
+      const storedSequences = storedReplaySessions[sessionId].sessionSequences;
+      for (const storedSeqId in storedSequences) {
+        const seq = storedSequences[storedSeqId];
+        const numericSeqId = parseInt(storedSeqId, 10);
+        const numericSessionId = parseInt(sessionId, 10);
+        if (numericSessionId === this.config.sessionId && numericSeqId === this.currentSequenceId) {
+          continue;
+        }
+        if (seq.events.length && seq.status === RecordingStatus.RECORDING) {
           this.sendEventsList({
-            events: storedReplayEvents.events,
-            sequenceId: storedReplayEvents.sequenceId,
-            sessionId: parseInt(sessionId, 10),
+            events: seq.events,
+            sequenceId: numericSeqId,
+            sessionId: numericSessionId,
           });
         }
       }
-      this.events = [];
-      const currentSessionStoredEvents = this.config.sessionId && storedReplaySessions[this.config.sessionId];
-      this.currentSequenceId = currentSessionStoredEvents ? currentSessionStoredEvents.sequenceId + 1 : 0;
-      void this.storeEventsForSession([], this.currentSequenceId);
     }
-    this.recordEvents();
   }
 
   recordEvents() {
     this.stopRecordingEvents = record({
       emit: (event) => {
         const eventString = JSON.stringify(event);
-        // Send the first two recorded events immediately
```

Comment:
I decided to get rid of this immediate send since we are batching events on the backend every 90 seconds, there is no point to immediately sending the first two events

Reply:
I think the other case for this was to avoid having dangling references in the regular events. I.e. the user opens browser, triggers some amplitude events then leaves, that would lead to missing recording error.

Reply:
Hmm that's true, I have the minimum interval set to 1 second right now, I could just reduce it even further? This specific logic was causing some weird edge cases so instead of fixing those I thought I'd just simplify the logic.